src/themes/default/pages/Product.vue

```diff
@@ -260,85 +266,84 @@ export default {
     Spinner,
     LazyHydrate
   },
-  mixins: [Product, VueOfflineMixin],
+  // Remove product.js dependency and use onlineHelper
+  mixins: [VueOfflineMixin, ProductOption],
   directives: { focusClean },
   data () {
     return {
       detailsOpen: false,
       quantity: 0,
-      isProductLoading: true
+      isProductLoading: false
     }
   },
   computed: {
+    ...mapGetters({
+      getCurrentCategory: 'category-next/getCurrentCategory',
+      getBreadcrumbs: 'category-next/getBreadcrumbs',
+      getCurrentProduct: 'product/getCurrentProduct',
+      getProductGallery: 'product/getProductGallery',
+      getCurrentProductConfiguration: 'product/getCurrentProductConfiguration',
+      getOriginalProduct: 'product/getOriginalProduct',
+      attributesByCode: 'attribute/attributeListByCode'
+    }),
     structuredData () {
       return {
-        availability: this.product.stock.is_in_stock ? 'InStock' : 'OutOfStock'
+        availability: this.getCurrentProduct.stock.is_in_stock ? 'InStock' : 'OutOfStock'
       }
     },
     getProductOptions () {
       if (
-        this.product.errors &&
-        Object.keys(this.product.errors).length &&
-        Object.keys(this.configuration).length
+        this.getCurrentProduct.errors &&
+        Object.keys(this.getCurrentProduct.errors).length &&
+        Object.keys(this.getCurrentProductConfiguration).length
       ) {
         return []
       }
-      return this.product.configurable_options
+      return this.getCurrentProduct.configurable_options
     },
-    getAvailableFilters () {
-      let filtersMap = {}
-      // TODO move to helper
-      if (this.product && this.product.configurable_options) {
-        this.product.configurable_options.forEach(configurableOption => {
-          const type = configurableOption.attribute_code
-          const filterVariants = configurableOption.values.map(
-            ({ value_index, label }) => {
-              let currentVariant = this.options[type].find(
-                config => config.id === value_index
-              )
-              label =
-                label || (currentVariant ? currentVariant.label : value_index)
-              return { id: value_index, label, type }
-            }
-          )
-          const availableOptions = filterVariants.filter(option =>
-            this.isOptionAvailable(option)
-          )
-          filtersMap[type] = availableOptions
-        })
+    offlineImage () {
+      return {
+        src: this.getThumbnail(this.getCurrentProduct.image, config.products.thumbnails.width, config.products.thumbnails.height),
+        error: this.getThumbnail(this.getCurrentProduct.image, config.products.thumbnails.width, config.products.thumbnails.height),
+        loading: this.getThumbnail(this.getCurrentProduct.image, config.products.thumbnails.width, config.products.thumbnails.height)
```

Comment:
shouldn't it be like `getOfflineImage`?

Reply:
i think it shouldn't. It's theme-specific and doing it this way we're allowing users to easily modify or even remove it.

Reply:
oh, you mean computed name?
i agree if so ;D

Reply:
yes, computer property name